core/src/test/scala/org/apache/spark/storage/BlockManagerDecommissionIntegrationSuite.scala

```diff
@@ -36,21 +37,34 @@ class BlockManagerDecommissionIntegrationSuite extends SparkFunSuite with LocalS
   val numExecs = 3
   val numParts = 3
 
+  test(s"verify that an already running task which is going to cache data succeeds " +
+    s"on a decommissioned executor after task start") {
+    runDecomTest(true, false, true, 1)
+  }
+
+  test(s"verify that an already running task which is going to cache data succeeds " +
+    s"on a decommissioned executor after iterator start") {
+    runDecomTest(true, false, true, 2)
+  }
+
```

Comment:
@holdenk .. this is what I meant: I have added two tests to test the decom after "Task Start". 

The first test is a negative test. It defines task start as when the driver says that the task is "launched" and as we discussed, it is expected to fail because decommissioning will be triggered before `BlockManager.doPut` is called and thus fail the task without starting the iterator.

The second test is what you want mentioned in your "intention": We want the decommissioning after the "iterator start" (ie user code started) to trigger a migration: The block manager will write the block anyway and it will be migrated.

The first negative test is just to demonstrate that it is a case we don't handle yet. Since we agree that it is not a case worth fixing for production, I will eventually delete it. 

So now we cover the full gamut of interesting decom points in task: When it is just launched, When user code is activated, After it has ended.